docker/Dockerfile.contrib

```
@@ -0,0 +1,31 @@
+FROM node AS build
+
+COPY node-zwave-js /root/node-zwave-js
+COPY zwavejs2mqtt /root/zwavejs2mqtt
+
+WORKDIR /root/node-zwave-js
+RUN npm install
+RUN npm run build
+
+WORKDIR /root/zwavejs2mqtt
+RUN npm config set unsafe-perm true
+RUN npm install
+RUN npm run build
+# RUN npm prune --production
+RUN cp -rL /root/node-zwave-js/packages/zwave-js node_modules/
```

```suggestion
RUN npm install ../node-zwave-js
```

This will create a symlink to the repository

---

Wouldn't it make more sense to `npm install ../node-zwave-js/packages/zwave-js`? Not sure though, this whole `npm workspace` link magic is also new to me :D

---

Shouldn't npm 7 do the job?

---

Otherwise, try both solutions @larstobi

---

I'm pretty newb with the npm tool, but so far, to me it is behaving weirdly. Especially with the dependency vendoring and the symlinking. 

And I'm getting a lot of "npm ERR! EISDIR: illegal operation on a directory, copyfile" messages in a lot of the npm commands.

```
Step 12/15 : RUN npm install ../node-zwave-js/packages/zwave-js
 ---> Running in a68b191e94a7
npm ERR! code EISDIR
npm ERR! syscall copyfile
npm ERR! path /home/node/zwavejs2mqtt/node_modules/agent-base
npm ERR! dest /home/node/zwavejs2mqtt/node_modules/.agent-base-lA1oG2lU
npm ERR! errno -21
npm ERR! EISDIR: illegal operation on a directory, copyfile '/home/node/zwavejs2mqtt/node_modules/agent-base' -> '/home/node/zwavejs2mqtt/node_modules/.agent-base-lA1oG2lU'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2020-11-26T15_46_33_668Z-debug.log
The command '/bin/sh -c npm install ../node-zwave-js/packages/zwave-js' returned a non-zero code: 235
```

I am spending a lot of time on strange npm errors, but hopefully I'm learning lots of new stuff. :-)

---

So, I figured this one is because zwavejs2mqtt already has agent-base as a dependency, so trying to install it again fails. Of course.

---

So it's fixed?

---

No, I don't know how to fix it. Will keep working on it, though.

---

Without the dependencies from zwave-js it won't run.

---

Did you try with `npm install ../node-zwave-js/packages/zwave-js`?